src/StarsRatingBar/StarsRatingBar.js

```diff
@@ -0,0 +1,239 @@
+    const isCurrentStarHovered = hoveredStarIndex === ratingValue;
+
+    // If the user hovers on a star the value should be compatible to the value of the hovered star
+    // otherwise the value should be compatible to the selected value.
+    const isFilledStar = isStarsHovered
+      ? ratingValue <= hoveredStarIndex
+      : ratingValue <= value;
+
+    const commonProps = {
+      key: ratingValue,
+      size: starRatingBarSizesInPx[starsRatingBarSize],
+      onMouseEnter: () => this._onMouseEnter(ratingValue),
+      onMouseLeave: () => this._onMouseLeave(),
+      onClick: () => this._onStarIconClick(ratingValue),
+    };
+
+    return isFilledStar ? (
+      <StarFilledIcon
+        {...commonProps}
+        data-hook={dataHooks.filledStar}
+        data-index={ratingValue}
+        {...styles(
+          'star',
+          { filled: true, hovered: isCurrentStarHovered },
+          this.props,
+        )}
+      />
+    ) : (
+      <StarIcon
+        {...commonProps}
+        data-index={ratingValue}
+        {...styles(
+          'star',
+          { empty: true, hovered: isCurrentStarHovered },
+          this.props,
+        )}
+      />
+    );
+  };
+
+  _shouldShowRateCaption = () => {
+    const { readOnly, rateCaptions } = this.props;
+    let shouldShowRateCaption = false;
+
+    if (rateCaptions) {
+      const isValidRateCaption =
+        Array.isArray(rateCaptions) && rateCaptions.length === 5;
+
+      if (readOnly) {
+        throw new Error('Rate caption is not available in read only mode.');
+      } else if (!isValidRateCaption) {
+        throw new Error('Rate caption must be an array of strings at size 5.');
+      }
+      shouldShowRateCaption = true;
+    }
+
+    return shouldShowRateCaption;
+  };
+
+  _renderRateCaption = () => {
+    const { rateCaptions, value } = this.props;
+    const { hoveredStarIndex } = this.state;
+    const isStarsHovered = hoveredStarIndex !== 0;
+
+    let rateCaptionCurrentLabel = '';
+
+    // If the user hovers on a star the label should be compatible to the value of the hovered star
+    // otherwise the label should be compatible to the selected value.
+    if (isStarsHovered) {
+      rateCaptionCurrentLabel = rateCaptions[hoveredStarIndex - 1];
+    } else {
+      rateCaptionCurrentLabel = value === 0 ? '' : rateCaptions[value - 1];
+    }
+
+    return (
+      <div className={styles.rateCaption}>
+        <Text
+          dataHook={dataHooks.ratingCaption}
+          ellipsis
+          size="small"
+          weight="bold"
+          secondary
+        >
+          {rateCaptionCurrentLabel}
+        </Text>
+      </div>
+    );
+  };
+
+  render() {
+    const { dataHook, className } = this.props;
+
+    return (
+      <div data-hook={dataHook} {...styles('root', {}, { className })}>
+        <div className={styles.starsContainer}>{this._renderStars()}</div>
+        {this._shouldShowRateCaption() ? this._renderRateCaption() : null}
+      </div>
```

Comment:
I think this can be a better layout.
What do you think?

```suggestion
      <Box data-hook={dataHook} {...styles('root', {}, { className })}>
        {this._renderStars()}
        {this._shouldShowRateCaption() ? this._renderRateCaption() : null}
      </Box>
```